pkg/handler/handler.go

```diff
@@ -202,3 +211,53 @@ func toGzip(data []byte) []byte {
 	zw.Close()
 	return buf.Bytes()
 }
+
+// RegisterOpenAPIVersionedService registers a handler to provides access to provided swagger spec.
+func RegisterOpenAPIVersionedService(openapiSpec *spec.Swagger, servePath string, handler common.PathHandler) (*OpenAPIService, error) {
+	o := OpenAPIService{}
+	if err := o.UpdateSpec(openapiSpec); err != nil {
+		return nil, err
+	}
+
+	files := map[string]func() ([]byte, string, time.Time){
+		"application/json":                                           o.getSwaggerBytes,
+		"application/com.github.proto-openapi.spec.v2@v1.0+protobuf": o.getSwaggerPbBytes,
```

Comment:
How the user show now this exact string for proto? we can document it but it would be nice if we can return list of supported types with the response, probably in the same Accept header. Also with that, we can support HEAD call too so a user just do the HEAD call and get the list and do the right call with the right type.

Reply:
I'm afraid there is no standardized way to do that kind of negotiation in HTTP. The closest thing I guess is `HTTP 300 Multiple Choices`, but it uses response body, URL to distinguish types, and there is no standardized way of choosing one of the responses on client side. We should probably document it.

Reply:
Documenting it looks good for now.